Xamarin.PropertyEditing.Mac/PropertyTableDelegate.cs

```diff
@@ -140,9 +140,15 @@ public override void ItemDidExpand (NSNotification notification)
 				return;
 			}
 
-			if (facade.Target is PanelGroupViewModel group)
+			if (facade.Target is PanelGroupViewModel group) {
 				this.dataSource.DataContext.SetIsExpanded (group.Category, isExpanded: true);
-			else if (facade.Target is ObjectPropertyViewModel ovm) {
+
+				// Also update the expander state. This is needed when the property editor first
+				// displays and category groups default to expanded, to show the right icon
+				NSView view = outline.GetView (0, row, makeIfNecessary: false);
```

Comment:
I don't think this is the right way to do it, you need to use `expandItem:` https://developer.apple.com/documentation/appkit/nsoutlineview/1532826-expanditem?language=objc

Reply:
The intention here isn't to expand the item. It's already been expanded, which is why ItemDidExpand is being called. This code is only there to update the icon so that the icon shows the right state (the expanded icon instead of the collapsed icon). Without it the icon is wrong when the item is expanded initially.

Reply:
@BretJohnson this is because you are using a custom expander? just to be in context

Reply:
@netonjm I'm not sure if you'd call this a custom expander - I didn't write this code originally. Would you mind taking a look at the code to see. The expand/collapse NSButton in the NSOutlineView looks to just be the default one, not particularly custom. However, there is lots of code there already to update the button State, so that must be needed; the code was just missing in the default case, where the item defaults to being expanded instead of the user explicitly expanding.

Reply:
A question I have is if ItemDidExpand needs this change ItemDidCollapse will need another patch in the same way?

Reply:
No. The icon defaults to collapsed, so when that happens by default the case is covered and if the user explicitly collapses, that's also covered by the existing code.

Reply:
I'm still not sure how we can get into a state where the outline view state and the UI are not synced.

Reply:
In this code, the icon for expand/collapse state is always set explicitly. The icon UI would never change if the code didn't update it. It must be a custom thing, where automatically updating the UI just doesn't happen otherwise. I'm not sure why that is as I didn't write this originally, but I expect it was done on purpose out of necessity. Given all that, the fix here is to explicitly set the icon in the one case that's not handled today - when it's expanded by default. After that, if the user expands/collapses the event notifications update the icon (via existing code in that class).

Reply:
https://github.com/xamarin/Xamarin.PropertyEditing/pull/520/commits/96561f19bf1c81e24c852e0a88ef46bd72e05e38

Reply:
That's a lot of code for things that have equivalent APIs. It may not have been the case in the past, but this can be done properly with less perf overhead and overall code.